### PR TITLE
Replace all uses of move_to_operands

### PR DESCRIPTION
**Pull request by tautschnig**
move_to_operands is deprecated, use add_to_operands(std::move(...)) or proper
constructors instead.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-symex/symex_builtin_functions.cpp [hunk 1]
```diff
@@ -434,10 +434,8 @@ void goto_symext::symex_cpp_new(
   exprt rhs(ID_address_of, pointer_type);
 
   if(do_array)
-  {
-    index_exprt index_expr(symbol.symbol_expr(), from_integer(0, index_type()));
-    rhs.move_to_operands(index_expr);
-  }
+    rhs.add_to_operands(
```
Comment (smowton) [inline src/goto-symex/symex_builtin_functions.cpp:437]: AFAIK usually keep braces when the if statement is multi-line, even if it's a single statement

**Review by smowton — APPROVED** [review]

File: src/goto-symex/symex_builtin_functions.cpp (hunk 1 above)
Comment (tautschnig) [reply src/goto-symex/symex_builtin_functions.cpp:437]: Fixed!

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1c5cab8f7fa6e7d139d2034b5411fdc9fac83a63).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96419499

File: src/cpp/cpp_typecheck.cpp [hunk 2]
```diff
@@ -196,7 +196,7 @@ void cpp_typecheckt::static_and_dynamic_initialization()
 
   dynamic_initializations.clear();
 
-  // block_sini.move_to_operands(block_dini);
+  // block_sini.add_to_operands(std::move(block_dini));
```
Comment (chrisr-diffblue) [inline src/cpp/cpp_typecheck.cpp]: Should this line just be deleted, given that it's commented out?

**Review by chrisr-diffblue — APPROVED** [review]

File: src/cpp/cpp_typecheck.cpp (hunk 2 above)
Comment (tautschnig) [reply src/cpp/cpp_typecheck.cpp]: Possibly so, but with the C++ front-end there is just a lot of cleanup to be done and there may be genuine value in some of the code currently commented out. So I'd rather leave this to be done in other PRs.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 85e8841ee570114a893f153e873fa0a6eaa2bf8e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96877361

File: src/cpp/cpp_typecheck.cpp (hunk 2 above)
Comment (peterschrammel) [reply src/cpp/cpp_typecheck.cpp]: I'd prefer using an ifdef 0 plus a comment that explains why the code is commented out/worth keeping.

**Review by peterschrammel — APPROVED** [review]

File: src/cpp/cpp_typecheck.cpp (hunk 2 above)
Comment (tautschnig) [reply src/cpp/cpp_typecheck.cpp]: It seems neither `block_sini` nor `block_dini` exist in anywhere in the code base, I have thus completely removed the line as it is impossible to make sense of it.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 471b95d4e86edf0b09a567e976e3701c824061e9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97275789

**Review by martin-cs — APPROVED** [review]
The parts that touch my parts of the code look fine.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b507534ad8f750a99fca8b75ff4929c53d019c11).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97521699

File: src/cpp/cpp_typecheck_constructor.cpp [hunk 3]
```diff
@@ -671,18 +667,13 @@ void cpp_typecheckt::full_member_initialization(
...
+          cpp_namet("@most_derived").as_expr(), std::move(tmp));
 
-        {
-          codet tmp(ID_member_initializer);
-          tmp.swap(final_initializers.get_sub().back());
-          cond.move_to_operands(tmp);
-          final_initializers.get_sub().back().swap(cond);
-        }
+        final_initializers.get_sub().back().swap(cond);
       }
```
Comment (kroening) [inline src/cpp/cpp_typecheck_constructor.cpp:677]: Subtil!

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b2eff7e347539167e3fee0b1a4b66a2806142f55).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97969473

**Pull request merged**